### PR TITLE
Add UserContextService bean for custom implementation of roles

### PR DESCRIPTION
**Pull request by testower**

### CODE REVIEW

File: src/main/java/no/entur/uttu/graphql/ProviderGraphQLSchema.java [hunk 1]
```diff
@@ -137,10 +150,29 @@ private void initCommonTypes() {
             .type(new GraphQLNonNull(codespaceObjectType))
         )
         .build();
+
+    userContextObjectType =
+      newObject()
+        .name("UserContext")
+        .field(
+          newFieldDefinition()
+            .name("preferredName")
```
Comment (vpaturet) [inline src/main/java/no/entur/uttu/graphql/ProviderGraphQLSchema.java:159]: missing description

File: src/main/java/no/entur/uttu/graphql/ProviderGraphQLSchema.java [hunk 2]
```diff
@@ -137,10 +150,29 @@ private void initCommonTypes() {
             .type(new GraphQLNonNull(codespaceObjectType))
         )
         .build();
+
+    userContextObjectType =
+      newObject()
+        .name("UserContext")
```
Comment (vpaturet) [inline src/main/java/no/entur/uttu/graphql/ProviderGraphQLSchema.java:156]: missing description

File: src/main/java/no/entur/uttu/graphql/ProviderGraphQLSchema.java [hunk 3]
```diff
@@ -137,10 +150,29 @@ private void initCommonTypes() {
...
+          newFieldDefinition()
+            .name("preferredName")
+            .type(new GraphQLNonNull(GraphQLString))
+        )
+        .field(
+          newFieldDefinition().name("isAdmin").type(new GraphQLNonNull(GraphQLBoolean))
+        )
+        .field(
+          newFieldDefinition()
+            .name("providers")
```
Comment (vpaturet) [inline src/main/java/no/entur/uttu/graphql/ProviderGraphQLSchema.java:167]: missing description

File: src/main/java/no/entur/uttu/security/UserContextService.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package no.entur.uttu.security;
+
+/**
+ *  This interface provides authorization features to the app
```
Comment (vpaturet) [inline src/main/java/no/entur/uttu/security/UserContextService.java:4]: The documentation could state that this abstraction can be used to extend/customize authorization features.

File: src/main/java/no/entur/uttu/security/UserContextService.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
+package no.entur.uttu.security;
```
Comment (vpaturet) [inline src/main/java/no/entur/uttu/security/UserContextService.java:1]: In OTP similar abstractions are moved to a dedicated "spi" package (for example no.entur.uttu.security.spi) to stress the fact that this interface can be used to extend the product and that there is some guarantee of stability for the contract.

**Comment by testower** [comment]
One more TODO: Currently, the ability to run uttu locally without security features is lost (with proflle "local"). Should be reintroduced with a separate config.

**Pull request merged**